src/Routes/HomePage/HomePage.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {axiosInstance} from "../../component/axios/axios";
+import { Loader } from "../../component";
+
+export function HomePage(){
+    const [isAuthenticated, setIsAuthenticated] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+    const [products, setProducts] = useState([]);
+    const navigate = useNavigate();
+  
+    useEffect(() => {
+      setIsLoading(true);
```

Comment:
All this logic can be added to the handleAuthentication function, there is no need for this to be inside the useEffect directly.

Reply:
I already push the changes that you suggested. Thank you!